src/telegram/ext/_utils/networkloop.py

```diff
@@ -50,14 +50,19 @@ async def network_retry_loop(
     stop_event: Optional[asyncio.Event] = None,
     is_running: Optional[Callable[[], bool]] = None,
     max_retries: int,
+    infinite_loop: bool,
 ) -> None:
     """Perform a loop calling `action_cb`, retrying after network errors.
 
-    Stop condition for loop:
+    Stop condition for loop in case of `infinite_loop` is True:
         * `is_running()` evaluates :obj:`False` or
-        * return value of `action_cb` evaluates :obj:`False`
         * or `stop_event` is set.
-        * or `max_retries` is reached.
+
+    Stop condition for loop in case of `infinite_loop` is False:
+        * a call to `action_cb` succeeds or
+        * `is_running()` evaluates :obj:`False` or
+        * or `stop_event` is set or
+        * `max_retries` is reached.
```

Comment:
Would avoid the doubling of conditions here
```suggestion
    Additional stop condition for loop in case of `infinite_loop` is False:
        * a call to `action_cb` succeeds or
        * `max_retries` is reached.
```